android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -2241,6 +2263,46 @@ public void onSettingsOptionSelected()
     startActivity(intent);
   }
 
+  private void startTrackRecording(boolean toast)
+  {
+    if (!LocationUtils.checkFineLocationPermission(this))
+    {
+      Logger.i(TAG, "Location permission not granted");
+      // This variable is a simple hack to re initiate the flow
+      // according to action of user. Calling it hack because we are avoiding
+      // creation of new methods by using this variable.
+      mLocationPermissionRequestedForRecording = true;
+      mLocationPermissionRequest.launch(new String[] { ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION });
```

Comment:
Is a coarse permission check required? Or only a fine permission check can be done?

Reply:
checking both would be convenient i guess. I have copied it from the implementation of navigation system since it is already implemented and tested.